plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -590,6 +590,10 @@ private void deleteTableColumnStatistics(String databaseName, String tableName,
     public void updatePartitionStatistics(Table table, String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
     {
         List<Partition> partitions = getPartitionsByNames(table.getDbName(), table.getTableName(), ImmutableList.of(partitionName));
+        if (partitions.isEmpty()) {
+            // no partition found to update
+            return;
```

Comment:
> Avoid providing misleading failure cause to the user when trying to drop the statistics for a partition which doesn't exist

 Why not improve the error message?

Reply:
Modified the code to throw an error when the specified partition is not being found.